extensions/wrapper/wrapper/src/main/java/de/sovity/edc/ext/wrapper/api/ui/UiResourceImpl.java

```diff
@@ -150,8 +158,22 @@ public IdResponseDto initiateCustomTransfer(InitiateCustomTransferRequest reques
     }
 
     @Override
-    public IdResponseDto terminateContractAgreement(String contractAgreementId, ContractTerminationRequest contractTerminationRequest) {
-        return null;
+    public IdResponseDto terminateContractAgreement(
+        String contractAgreementId,
+        @Valid ContractTerminationRequest contractTerminationRequest) {
```

Comment:
In Quarkus you have a complete parsing and weaving of CDI units during compilation, in Spring you have a Proxy + AOP at runtime approach.

Here we have nothing, so we need to call the validation manually. We should also omit the `@Valid`. It's part of the validation API JAR, but will only do something if you have a well-integrated implementation in your stack. 

I don't think we can hack that right, unless the Eclipse EDC has something prepared for that, but since they don't even use JSON, but custom non-semantic JSON-LD parsing, I don't think they'll have something for us